### PR TITLE
Selecting rows with empty blobs throws exception

### LINKED ISSUE #6879: Selecting rows in Cassandra with empty blobs throws exception.
I'm using the Cassandra connector and Presto version 0.147. 

In Cassandra I have a table with a blob column in which some rows are set to an empty buffer value. For example,

` 
column_name
-----
  0x
`

If I run a select statement that includes that row I get the following exception: 

`java.lang.IllegalArgumentException: Invalid size: 0
	at io.airlift.slice.Slice.<init>(Slice.java:245)
	at io.airlift.slice.Slices.wrappedBuffer(Slices.java:120)
	at com.facebook.presto.cassandra.CassandraType.getColumnValue(CassandraType.java:191)
	at com.facebook.presto.cassandra.CassandraType.getColumnValue(CassandraType.java:151)
	at com.facebook.presto.cassandra.CassandraRecordCursor.getSlice(CassandraRecordCursor.java:122)
	at com_facebook_presto_$gen_CursorProcessor_132.project_0(Unknown Source)
	at com_facebook_presto_$gen_CursorProcessor_132.process(Unknown Source)
	at com.facebook.presto.operator.ScanFilterAndProjectOperator.getOutput(ScanFilterAndProjectOperator.java:215)
	at com.facebook.presto.operator.Driver.processInternal(Driver.java:378)
	at com.facebook.presto.operator.Driver.processFor(Driver.java:301)
	at com.facebook.presto.execution.SqlTaskExecution$DriverSplitRunner.processFor(SqlTaskExecution.java:618)
	at com.facebook.presto.execution.TaskExecutor$PrioritizedSplitRunner.process(TaskExecutor.java:529)
	at com.facebook.presto.execution.TaskExecutor$Runner.run(TaskExecutor.java:665)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)

2016-12-14T03:03:32.145Z	ERROR	remote-task-callback-4186	com.facebook.presto.execution.StageStateMachine	Stage 20161214_030331_00047_e2v33.1 failed
java.lang.IllegalArgumentException: Invalid size: 0
	at io.airlift.slice.Slice.<init>(Slice.java:245)
	at io.airlift.slice.Slices.wrappedBuffer(Slices.java:120)
	at com.facebook.presto.cassandra.CassandraType.getColumnValue(CassandraType.java:191)
	at com.facebook.presto.cassandra.CassandraType.getColumnValue(CassandraType.java:151)
	at com.facebook.presto.cassandra.CassandraRecordCursor.getSlice(CassandraRecordCursor.java:122)
	at com_facebook_presto_$gen_CursorProcessor_132.project_0(Unknown Source)
	at com_facebook_presto_$gen_CursorProcessor_132.process(Unknown Source)
	at com.facebook.presto.operator.ScanFilterAndProjectOperator.getOutput(ScanFilterAndProjectOperator.java:215)
	at com.facebook.presto.operator.Driver.processInternal(Driver.java:378)
	at com.facebook.presto.operator.Driver.processFor(Driver.java:301)
	at com.facebook.presto.execution.SqlTaskExecution$DriverSplitRunner.processFor(SqlTaskExecution.java:618)
	at com.facebook.presto.execution.TaskExecutor$PrioritizedSplitRunner.process(TaskExecutor.java:529)
	at com.facebook.presto.execution.TaskExecutor$Runner.run(TaskExecutor.java:665)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)

`

It seems like I should be able to run this query successfully.

### PR DESCRIPTION
**Pull request by delding**
FIX #6879 

Now querying a row with empty blobs will display nothing for the empty blob cell. Here are a visual compare of presto-cli and cqlsh

image

image

### CODE REVIEW

File: presto-main/etc/config.properties [hunk 1]
```diff
@@ -36,7 +36,8 @@ plugin.bundles=\
   ../presto-tpch/pom.xml, \
   ../presto-local-file/pom.xml, \
   ../presto-mysql/pom.xml,\
-  ../presto-postgresql/pom.xml
+  ../presto-postgresql/pom.xml,\
+  ../presto-cassandra/pom.xml
```
Comment (electrum) [inline presto-main/etc/config.properties]: Please remove this. We don't configure any Cassandra catalogs, so this just adds startup time needlessly.

**Comment by electrum** [comment]
Can you add a test for this?

**Comment by delding** [comment]
Sure, I'm adding test

File: presto-cassandra/src/main/java/com/facebook/presto/cassandra/CassandraType.java [hunk 2]
```diff
@@ -202,7 +203,11 @@ public static NullableValue getColumnValue(Row row, int i, CassandraType cassand
                     return NullableValue.of(nativeType, utf8Slice(row.getVarint(i).toString()));
                 case BLOB:
                 case CUSTOM:
-                    return NullableValue.of(nativeType, wrappedBuffer(row.getBytesUnsafe(i)));
+                    ByteBuffer value = row.getBytesUnsafe(i);
```
Comment (raghavsethi) [inline presto-cassandra/src/main/java/com/facebook/presto/cassandra/CassandraType.java]: This needlessly checks for array - we only care about the size of the buffer. Whether it's array backed or not is irrelevant.

Comment (delding) [inline presto-cassandra/src/main/java/com/facebook/presto/cassandra/CassandraType.java]: addressed

**Comment by delding** [comment]
Hi @raghavsethi , Test for `getColumnValue` needs to pass a Row object with a column of Blob type, but the cassandra `RowUtil` only provide `createSingleStringRow` method that creates a row with a column of String type. I can't think of a simpler way to test than using a MockRow as I mentioned to you earlier..

File: presto-cassandra/src/main/java/com/facebook/presto/cassandra/CassandraType.java (hunk 2 above)
Comment (raghavsethi) [inline presto-cassandra/src/main/java/com/facebook/presto/cassandra/CassandraType.java]: Remaining is the wrong choice - read the javadoc.

**Comment by delding** [comment]
updated and add a test

File: presto-cassandra/src/test/java/com/facebook/presto/cassandra/CassandraTestingUtils.java [hunk 3]
```diff
@@ -148,6 +150,17 @@ public static void createTableAllTypesPartitionKey(Session session, String keysp
         insertTestData(session, keyspace, tableName, date);
     }
 
+    public static void createTableWithEmptyBlobCell(Session session, String keyspace, String tableName, Date date)
+    {
+        session.execute("DROP TABLE IF EXISTS " + tableName);
```
Comment (electrum) [inline presto-cassandra/src/test/java/com/facebook/presto/cassandra/CassandraTestingUtils.java]: Put this inside the test method in `TestCassandraIntegrationSmokeTest`. Splitting this over two classes makes it a lot harder to read.

**Comment by delding** [comment]
@electrum pr updated

**Comment by raghavsethi** [comment]
Closing, since this has too many issues and needs to be re-done.

**Pull request closed without merging**